smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestResponseGenerator.kt

```diff
@@ -67,6 +57,26 @@ open class HttpProtocolUnitTestResponseGenerator protected constructor(builder:
         writer.write("}")
     }
 
+    private fun renderBuildHttpResponseParams(test: HttpResponseTestCase) {
+        val params = mutableListOf<String>()
+        params.add("code: ${test.code}")
+        if (test.headers.isNotEmpty()) {
+            var headers = mutableListOf<String>()
+            for (hdr in test.headers.entries) {
+                headers.add("    \"${hdr.key}\": \"${hdr.value}\"")
+            }
+            val headersString = "headers: [\n${headers.joinToString(",\n")}\n]"
```

Comment:
1.  nit: it would be better to abstract this out to a separate function
2. THANK YOU for doing this! wow, so much better!

Reply:
1. moved to a separate function
2. no prob, yay!